### ISSUE TITLE
ruff binary not found when installed to PATH

### ISSUE THREAD

**Issue by jaraco** [businho/pytest-ruff#21]
pytest-ruff seems to be making some assumptions about how binaries can be found for an environment, assumptions that fail expectations for a pip-run environment:

```
 ~ @ pip-run pytest-ruff -- -c "import pytest_ruff; print(pytest_ruff.find_ruff_bin())"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/var/folders/f2/2plv6q2n7l932m2x004jlw340000gn/T/pip-run-jrfhl2p1/ruff/__main__.py", line 28, in find_ruff_bin
    raise FileNotFoundError(path)
FileNotFoundError: /Users/jaraco/Library/Python/3.12/bin/ruff
```

Even though the binary is present and made prominent by the installation of pytest-ruff:

```
 ~ @ pip-run pytest-ruff -- -c "import shutil; print(shutil.which('ruff'))"
/var/folders/f2/2plv6q2n7l932m2x004jlw340000gn/T/pip-run-740q7ysr/bin/ruff
```

The same issue can be observed without using pip-run by using `pip install --target`:

```
 draft @ pip install --target target pytest-ruff -q
 draft @ env PYTHONPATH=target PATH=target/bin:$PATH py -c "import pytest_ruff; pytest_ruff.find_ruff_bin()"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/jaraco/draft/target/ruff/__main__.py", line 28, in find_ruff_bin
    raise FileNotFoundError(path)
FileNotFoundError: /Users/jaraco/Library/Python/3.12/bin/ruff
```

I understand that it may be undesirable to simply rely on `ruff` as found on $PATH (as `shutil.which` does), but it would be nice if ruff could be picked up if installed using `--target`.

**Comment by jaraco** [comment]
If you'd like me to explore a solution, feel free to assign the issue to me.

**Comment by iurisilvio** [comment]
It is not clear to me why pip-run can't find ruff, I just call ruff discovery function, but I agree if `FileNotFoundError`, we can check if `ruff` is available.

I'll assign it to you, I won't have time to work on it now.

**Comment by jaraco** [comment]
Oh, shoot. I see now that `find_ruff_bin` is imported from `ruff` so the discovery issue is in ruff itself.

**Comment by jaraco** [comment]
The issue was previously reported as https://github.com/astral-sh/ruff/issues/11246.

**Comment by jaraco** [comment]
Good news. The issue is fixed upstream for ruff > 0.4.4, so there's nothing more to do here.
